src/components/Composer/index.js

```diff
@@ -192,6 +201,54 @@ class Composer extends React.Component {
         this.textInput.removeEventListener('wheel', this.handleWheel);
     }
 
+    // get characters from the cursor to the next space or new line
+    getNextChars(str, cursorPos) {
+        // Get the substring starting from the cursor position
+        const substr = str.substring(cursorPos);
+
+        // Find the index of the next space or new line character
+        const spaceIndex = substr.search(/[ \n]/);
+
+        if (spaceIndex === -1) {
+            return substr;
+        }
+
+        // If there is a space or new line, return the substring up to the space or new line
+        return substr.substring(0, spaceIndex);
+    }
+
+    addCursorPositionToSelectionChange(event) {
+        if (this.props.shouldCalculateCaretPosition) {
+            const newValueBeforeCaret = event.target.value.slice(0, event.nativeEvent.selection.start);
+
+            this.setState(
+                {
+                    valueBeforeCaret: newValueBeforeCaret,
+                    caretContent: this.getNextChars(this.props.value, event.nativeEvent.selection.start),
+                },
+
+                () => {
+                    const customEvent = {
+                        nativeEvent: {
+                            selection: {
+                                start: event.nativeEvent.selection.start,
+                                end: event.nativeEvent.selection.end,
+
+                                // 4 is a width of the space which we added manually
```

Comment:
Should we add this as a constant?

Reply:
not sure if itwould be used anywhere else but i added it, thanks